File: algorithms/interview-preparation-kit-2/php/code/spec/PokemonCollection/PokemonCollection.spec.php
<?php

use PokemonCollection\PokemonCollection;

describe('Pokemon Collection', function () {
  xit('collection with 1 pokemon not including repetitions', function () {
    $pokemons = ['Zubat'];
    expect(PokemonCollection::notCaptured($pokemons))->toBe(150);
  });

  xit('collection with 1 pokemon including repetitions', function () {
    $pokemons = ['Zubat', 'Zubat', 'Zubat', 'Zubat', 'Zubat', 'Zubat', 'Zubat', 'Zubat'];
    expect(PokemonCollection::notCaptured($pokemons))->toBe(150);
  });

  xit('collection with 2 pokemon including repetitions', function () {
    $pokemons = ['Zubat', 'Pikachu', 'Pikachu'];
    expect(PokemonCollection::notCaptured($pokemons))->toBe(149);
  });

  xit('collection with 4 pokemon not including repetitions', function () {
    $pokemons = ['Zubat', 'Charmander', 'Caterpie', 'Pidgeot'];
    expect(PokemonCollection::notCaptured($pokemons))->toBe(147);
  });

  xit('collection with 5 pokemon including repetitions', function () {
    $pokemons = ['Charmander', 'Caterpie', 'Pidgeot', 'Rattata', 'Zubat', 'Zubat', 'Zubat'];
    expect(PokemonCollection::notCaptured($pokemons))->toBe(146);
  });
});

File: algorithms/interview-preparation-kit-2/ruby/code/spec/PokemonCollection/pokemon_collection.spec.rb
require_relative "../../src/PokemonCollection/pokemon_collection"

describe "Pokemon Collection" do
  context "collection with 1 pokemon not including repetitions" do
    it "it 150" do
      pokemons = ['Zubat'];
      expect(PokemonCollection.notCaptured(pokemons)).to eq(150)
    end
  end    

  xcontext "collection with 1 pokemon including repetitions" do
    it "it 150" do
      pokemons = ['Zubat', 'Zubat', 'Zubat', 'Zubat', 'Zubat', 'Zubat', 'Zubat', 'Zubat'];
      expect(PokemonCollection.notCaptured(pokemons)).to eq(150)
    end
  end    

  xcontext "collection with 2 pokemon including repetitions" do
    it "it 149" do
      pokemons = ['Zubat', 'Pikachu', 'Pikachu'];
      expect(PokemonCollection.notCaptured(pokemons)).to eq(149)
    end
  end    

  xcontext "collection with 4 pokemon not including repetitions" do
    it "it 147" do
      pokemons = ['Zubat', 'Charmander', 'Caterpie', 'Pidgeot'];
      expect(PokemonCollection.notCaptured(pokemons)).to eq(147)
    end
  end    

  xcontext "collection with 5 pokemon including repetitions" do
    it "it 146" do
      pokemons = ['Charmander', 'Caterpie', 'Pidgeot', 'Rattata', 'Zubat', 'Zubat', 'Zubat'];
      expect(PokemonCollection.notCaptured(pokemons)).to eq(146)
    end
  end    
end

